airflow/dags/oncokb_curation.py

```diff
@@ -0,0 +1,136 @@
+from datetime import timedelta, datetime
+from airflow.decorators import dag, task
+from airflow.io.path import ObjectStoragePath
+import pandas as pd
+import logging
+
+#Read data locally, we can change it to msk API later
+local = "path/to/folder/with/msk/data"
+base = ObjectStoragePath("s3://my_aws_s3@oncokb-ge")
+filter_path = "path/to/data"
+
+default_args = {
+    	'owner': 'oncoKB',
+		'retries': 3,
+		'retry_delay': timedelta(minutes=5)
+}
+
+@dag(
+    default_args=default_args,
+    dag_id="oncokb_curation_v1",
+    description="Get the statistic file and save to S3",
+    start_date=datetime(2024,6,8,0),
+    schedule_interval="@weekly",
+    catchup=False,
+)
+
+def curate_mutation_data():
+
+    def exclude_samples(data,filter_path):
+        excluded_sample_ids = pd.read_csv(filter_path,delimiter='\t',header=None)[0].tolist()
+        filtered_data = data[~data['SAMPLE_ID'].isin(excluded_sample_ids)]
+        return filtered_data
+    
+    def upload_to_s3(path,df):
+        with path.open("w") as file:
+            df.to_csv(file, sep='\t', index=False)
+
+    def read_from_s3(path: ObjectStoragePath):
+        with path.open() as f:
+            df = pd.read_csv(f,sep='\t')
+        return df
+
+    @task
+    def get_statistic_file():
+        test_name="msk_impact_2017"
+        file_path = local+test_name
+
+        # get the mutation data
+        sample = pd.read_csv(file_path+"/data_clinical_sample.txt", usecols=['SAMPLE_ID', 'CANCER_TYPE','CANCER_TYPE_DETAILED'], delimiter='\t', skiprows=4)
+        mutation = pd.read_csv(file_path+"/data_mutations.txt", usecols=['Hugo_Symbol', 'Tumor_Sample_Barcode','HGVSp_Short'], delimiter='\t',skiprows=1)
+        data = pd.merge(sample, mutation, left_on='SAMPLE_ID', right_on='Tumor_Sample_Barcode').drop(columns=['Tumor_Sample_Barcode'])
+        if filter_path:
+            data = exclude_samples(data,filter_path)
+
+        #all cancer frequency
+        all_number = data[['Hugo_Symbol', 'HGVSp_Short']].value_counts().reset_index(name='Number')
+        all_percentage = data[['Hugo_Symbol', 'HGVSp_Short']].value_counts(normalize=True).reset_index(name='Percentage')
+        all_frequency = pd.merge(all_number, all_percentage, on=['Hugo_Symbol', 'HGVSp_Short'])
+        all_path = base / f"{test_name}_all_frequency.txt"
+        upload_to_s3(all_path,all_frequency)
+        logging.info("upload all_frequecny file")
+
+        #cancer type frequency
+        type_number = data.groupby('CANCER_TYPE')[['Hugo_Symbol', 'HGVSp_Short']].value_counts().rename("Number").reset_index()
```

Comment:
Just curious, why are you doing `.rename("Number").reset_index()` here and `.reset_index(name='Number')` in the previous section?